v2/protocol/http/protocol.go

```diff
@@ -276,7 +279,9 @@ func (p *Protocol) Respond(ctx context.Context) (binding.Message, protocol.Respo
 		if !ok {
 			return nil, nil, io.EOF
 		}
-
+		if &in != nil && in.err != nil && strings.Contains(in.err.Error(), connectionReset) {
```

Comment:
just `in` not `&in`

Reply:
that gives me a type mismatch error as `in` is a struct of `msgErr` I can change it to comparing to an empty struct like `in != msgErr{}`